common/djangoapps/entitlements/models.py

```diff
@@ -302,6 +306,76 @@ def get_active_entitlements_for_user(cls, user):
             enrollment_course_run=None
         ).select_related('user').select_related('enrollment_course_run')
 
+    @classmethod
+    def does_user_have_active_entitlements(cls, user):
+        """
+        Checks whether or not a user has any active entitlements
+        """
+        count = cls.objects.count(
+            user=user
+        ).exclude(
+            expired_at__isnull=False,
+            enrollment_course_run=None
+        ).select_related('user').select_related('enrollment_course_run')
+        return count > 0
+
+    @classmethod
+    def get_fullfillable_entitlement_for_user(cls, user, course_run_key):
+        """
+        Looks at the User's existing entitlements to see if the user already has a Course Entitlement for the
+        course run provided in the course_key.  If the user does have an Entitlement with no run set, the User is
+        enrolled in the mode set in the Entitlement.
+
+        Note: This method assumes that eligibility to enroll in the course has already been checked.
+
+        Arguments:
+            user (User): The user that we are inspecting the entitlements for.
+            course_run_key (CourseKey): The course run Key.
+
+        Returns:
+            CourseEntitlement: The most recent fullfillale CourseEntitlement, None otherwise.
+        """
+        # Check if the User has any Entitlements that are active
+        if cls.does_user_have_active_entitlements(user):
+            course_uuid = get_course_uuid_for_course(course_run_key)
+            if course_uuid:
+                # Look for an entitlement for the current uuid
+                try:
```

Comment:
This try except block might be something worth pulling into 'get_fullfillable_entitlements_for_user()' which could take an optional course param too. something like:
```python
def get_fullfillable_entitlements_for_user(user, course_uuid)
  try:
    course_entitlements = cls.objects.filter(
      user=user,
      course_uuid=course_uuid
    ).exclude(
      expired_at__isnull=False,
      enrollment_course_run__isnull=False
    )
    return course_entitlements
  except CourseEntitlement.DoesNotExist:
    return None
```

Not sure if it would be best to include the course_uuid look up in that function so that it could take a course_key param instead of a uuid.
Alternatively it could just return all fullfillable entitlements for a user and you could filter that for an entitlement for the specific run.